### DISCUSSION TITLE
Docker error 500 : ValueError: 'unknown' does not appear to be an IPv4 or IPv6 address

### DISCUSSION THREAD

**Discussion by keyermann** [healthchecks/healthchecks#1086] [General]
Hi everyone,

I am currently testing HeathChecks via Docker to see if it meets our needs.
But I systematically run into an error I can't identify the cause.

Every time I try to ping I get a 500 error with the message: ValueError: 'unknown' does not appear to be an IPv4 or IPv6 address

Is this a configuration error?

**Comment by cuu508** [comment]
For every ping, Healthchecks logs the client's IP address in the database.

To determine client's IP address, it first looks at the `X-Forwarded-For` request header. If it is empty, it uses `request.META.REMOTE_ADDR` (the IP address of the client). 

Code: https://github.com/healthchecks/healthchecks/blob/27ecb40c889ebdb918e25663a9d1e798791359a7/hc/api/views.py#L187

I'm guessing the `X-Forwarded-For` header contains the string `unknown`, which does not validate as an IP address.

**Reply by keyermann** [reply]
Thanks for your reply.
You must be right as when I add manualy an X-Forwarded-For IP in the header request it works.

Could it be cause of my company proxy ?
Is there a way for me to get around this without having to manually add the `X-Forwarded-For` header every time?

**Reply by cuu508** [reply]
If you are planning to run Healthchecks in production, the best practice is to run it behind a reverse proxy. The reverse proxy then can be configured to set its own `X-Forwarded-For` header.

**Reply by keyermann** [reply]
Ok thank you for you help
